daemon/src/taker_cfd.rs

```diff
@@ -191,6 +193,31 @@ where
 
         let order_to_take = order_to_take.context("Order to take could not be found in current maker offers, you might have an outdated offer")?;
 
+        // Construct the CFD so we can retrieve the margin
+        let cfd = Cfd::from_order(
+            &order_to_take,
+            quantity,
+            self.maker_identity,
+            Some(self.maker_peer_id),
+            Role::Taker,
+            leverage,
+        );
+
+        // Make sure we have enough balance to open this position
+        let balance = self
+            .wallet
+            .send(wallet::GetBalance)
+            .await
+            .context("Wallet actor unavailable")??;
+        let my_margin = cfd.margin();
```

Comment:
Probably impossible to action, but it's interesting to note that the margin doesn't take into account the transaction fee, which is calculated later on.

This means that we could sometimes succeed on the following check and still not have enough money to pay for the lock transaction.

Reply:
It is actually possible! This inspired me to give it another iteration :)